tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,365 @@
+        assert os.path.exists(workspace_path)
+        assert os.path.exists(os.path.join(workspace_path, "configuration.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "data.yaml"))
+        template = MPAConfig.fromfile(os.path.join(workspace_path, "template.yaml"))
+        expected_template_train_type = {"default_value": "SELFSUPERVISED"}
+        assert template.hyper_parameters.parameter_overrides.algo_backend.train_type == expected_template_train_type
+        model_dir = os.path.join(workspace_path, train_type)
+        assert os.path.exists(model_dir)
+        assert os.path.exists(os.path.join(model_dir, "model.py"))
+        assert os.path.exists(os.path.join(model_dir, "data_pipeline.py"))
+
+        # 4. Raising ValueError with wrong train_type
+        workspace_path = os.path.join(tmp_dir_path, "otx_cli_builder_wrong")
+        train_type = "unexpected"
+        inputs = {
+            "task_type": "classification",
+            "train_type": train_type,
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        with pytest.raises(ValueError):
+            otx_builder.build_task_config(**inputs)
+
+        # 5. Build workspace with model_type argments
+        workspace_path = os.path.join(tmp_dir_path, "otx_cli_builder_yolox")
+        model_type = "yolox"
+        inputs = {
+            "task_type": "detection",
+            "model_type": model_type,
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        otx_builder.build_task_config(**inputs)
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        template = MPAConfig.fromfile(os.path.join(workspace_path, "template.yaml"))
+        assert template.name.lower() == model_type
+
+        # 6. Raise ValueError when build workspace with wrong model_type argments
+        workspace_path = os.path.join(tmp_dir_path, "otx_cli_builder_atss_2")
+        inputs = {
+            "task_type": "detection",
+            "model_type": "unexpected",
+            "workspace_path": workspace_path,
+            "otx_root": otx_root,
+        }
+        with pytest.raises(ValueError):
+            otx_builder.build_task_config(**inputs)
+
+    @e2e_pytest_unit
+    def test_builder_build_backbone_config(self, tmp_dir_path):
+        """
+        <b>Description:</b>
+        Check "Builder.build_backbone_config" function that generate backbone configuration file is working well
+
+        <b>Steps</b>
+        1. Generate backbone config file (mmcls.MMOVBackbone)
+        2. Raise ValueError with wrong output_path
+        """
+        otx_builder = Builder()
+
+        # 1. Generate backbone config file (mmcls.MMOVBackbone)
+        tmp_backbone_path = os.path.join(tmp_dir_path, "backbone.yaml")
+        backbone_type = "mmcls.MMOVBackbone"
```

Comment:
Instead of hardcoding it, could we move this variable with pytest.mark.parametrize decorator? It'd be easier to extend if we have different backbone types in the future.

```py
@pytest.mark.parametrize("tmp_dir_path, backbone", [tmp_dir_path, "mmcls.MMOVBackbone"] )
def test_builder_build_backbone_config(self, tmp_dir_path: str, backbone: str):
```

Reply:
It looks good. I will apply this for the future. :)

Reply:
Reflected on this, https://github.com/openvinotoolkit/training_extensions/pull/1519/commits/73bb47915e2071fbbcde0174c3087588032330c5